docs/AUTHENTICATION.md

```diff
@@ -178,3 +178,168 @@ EOF
+  name:  ${KUBERNETES_SERVICE_ACCOUNT}
+  namespace: ${NAMESPACE}
+---
+kind: Role
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: ${KUBERNETES_SERVICE_ACCOUNT}-sync
+rules:
+- apiGroups: [""]
+  resources:
+  - secrets
+  verbs:
+  - get
+  - create
+  - patch
+---
+kind: RoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: ${KUBERNETES_SERVICE_ACCOUNT}-sync-rb
+subjects:
+- kind: ServiceAccount
+  name: ${KUBERNETES_SERVICE_ACCOUNT}
+roleRef:
+  kind: Role
+  name: ${KUBERNETES_SERVICE_ACCOUNT}-sync
+  apiGroup: ""
+---
+apiVersion: batch/v1
+kind: CronJob
+metadata:
+  name: ${KUBERNETES_SERVICE_ACCOUNT}-credentials-sync
+spec:
+  suspend: false
+  schedule: "*/45 * * * *"
+  failedJobsHistoryLimit: 1
+  successfulJobsHistoryLimit: 1
+  concurrencyPolicy: Forbid
+  startingDeadlineSeconds: 1800
+  jobTemplate:
+    spec:
+      template:
+        spec:
+          serviceAccountName:  ${KUBERNETES_SERVICE_ACCOUNT}
+          restartPolicy: Never
+          containers:
+            - image: google/cloud-sdk:debian_component_based
+              name: create-access-token
+              imagePullPolicy: IfNotPresent
+              livenessProbe:
+                exec:
+                  command:
+                  - gcloud
+                  - version
+              readinessProbe:
+                exec:
+                  command:
+                  - gcloud
+                  - version
+              env:
+                - name: SECRET_NAME
+                  value: ${SECRET_NAME}
+                - name: SECRET_KEY
+                  value: ${SECRET_KEY}
+              command:
+                - /bin/bash
+                - -ce
+                - |-
+                  kubectl create secret generic $SECRET_NAME \
+                    --dry-run=client \
+                    --from-literal=$SECRET_KEY=$(gcloud auth print-access-token) \
+                    -o yaml | kubectl apply -f -
+              resources:
+                requests:
+                  cpu: 250m
+                  memory: 256Mi
+                limits:
+                  cpu: 500m
+                  memory: 512Mi
+EOF
+```
+Grant `roles/iam.workloadIdentityUser` to the GCP service account:
+
+```console
+$ gcloud iam service-accounts add-iam-policy-binding \
+    ${GCP_SERVICE_ACCOUNT}@${PROJECT_ID}.iam.gserviceaccount.com \
+    --role roles/iam.workloadIdentityUser \
+    --member "serviceAccount:${PROJECT_ID}.svc.id.goog[${NAMESPACE}/${KUBERNETES_SERVICE_ACCOUNT}]" \
+    --project ${PROJECT_ID}
+```
+
+Annotate the `ServiceAccount` with the email address of the GCP service account:
+
+```console
+$ kubectl annotate serviceaccount ${KUBERNETES_SERVICE_ACCOUNT} \
+    iam.gke.io/gcp-service-account=${GCP_SERVICE_ACCOUNT}@${PROJECT_ID}.iam.gserviceaccount.com \
+    -n ${NAMESPACE}
+```
+
+#### 3. Create initial Access Token
+```console
+kubectl create job --from=cronjob/${KUBERNETES_SERVICE_ACCOUNT}-credentials-sync cred-sync-001
```

Comment:
I fixed locally the error caused by the missing `namespace` field in the Role, RoleBinding and CronJob resources. After that, the `Pod` created by the `Job` was raising some errors (Warning Events), is this expected? That's the message I received from the `Pod`:

```console
Events:
  Type     Reason       Age                    From               Message
  ----     ------       ----                   ----               -------
  Normal   Scheduled    7m43s                  default-scheduler  Successfully assigned test-review/cred-sync-001--1-nwgq8 to k3d-review-brad-server-0
  Normal   Pulling      7m44s                  kubelet            Pulling image "google/cloud-sdk:debian_component_based"
  Normal   Pulled       6m39s                  kubelet            Successfully pulled image "google/cloud-sdk:debian_component_based" in 1m4.844241766s
  Normal   Created      6m34s                  kubelet            Created container create-access-token
  Normal   Started      6m33s                  kubelet            Started container create-access-token
  Warning  Unhealthy    6m33s                  kubelet            Readiness probe errored: rpc error: code = NotFound desc = failed to exec in container: failed to load task: no running task found: task 9f606fa70e0e2d2df503e59495dfff434c814a054124e2e233db3711b0822b0d not found: not found
  Warning  FailedMount  6m30s (x3 over 6m32s)  kubelet            MountVolume.SetUp failed for volume "kube-api-access-sjmcz" : object "test-review"/"kube-root-ca.crt" not registered
```

But after these errors, the `Job` completed successfully and the `Secret` was created as expected:
```console
➜ k get job -n test-review                           
NAME            COMPLETIONS   DURATION   AGE
cred-sync-001   1/1           72s        13m
➜ k get po -n test-review 
NAME                     READY   STATUS      RESTARTS   AGE
cred-sync-001--1-nwgq8   0/1     Completed   0          13m
➜ k get secret -n test-review                        
NAME                          TYPE                                  DATA   AGE
default-token-p2xc5           kubernetes.io/service-account-token   3      27m
token-generator-token-kj7sm   kubernetes.io/service-account-token   3      24m
gcp-credentials               Opaque                                1      12m
```

Reply:
See above